agnpy/tests/test_spectra.py

```diff
@@ -407,3 +484,82 @@ def test_from_norm_at_gamma_1(self):
             gamma_max=gamma_max_test,
         )
         assert u.isclose(norm, epwl(1), atol=0 * u.Unit("cm-3"), rtol=1e-2)
+
+class TestInterpolation:
+
+    """ 1. assert that outside the bounding box the function returns 0"""
+
+    def test_call_pow(self):
+        gamma = np.logspace(0, 8)
+        values = pwl_inter(gamma).value
+        condition = (gamma_min_test <= gamma) * (gamma <= gamma_max_test)
+        # check that outside the boundaries values are all 0
+        assert not np.all(values[~condition])
+
+    def test_call_bpwl(self):
+        gamma = np.logspace(0, 8)
+        values = bpwl_inter(gamma).value
+        condition = (gamma_min_test <= gamma) * (gamma <= gamma_max_test)
+        # check that outside the boundaries values are all 0
+        assert not np.all(values[~condition])
+
+    def test_call_lp(self):
+        gamma = np.logspace(0, 8)
+        values = lp_inter(gamma).value
+        condition = (gamma_min_test <= gamma) * (gamma <= gamma_max_test)
+        # check that outside the boundaries values are all 0
+        assert not np.all(values[~condition])
+
+    def test_call_ep(self):
+        gamma = np.logspace(0, 8)
+        values = epwl_inter(gamma).value
+        condition = (gamma_min_test <= gamma) * (gamma <= gamma_max_test)
+        # check that outside the boundaries values are all 0
+        assert not np.all(values[~condition])
+
+    """ 2. assert that the interpolated function does not have large deviations from the original one """
+
+    def test_power_vs_inter(self):
+        a = abs(pwl_inter(pwl_data[0,:]).value - pwl_data[1,:])/pwl_data[1,:]
+        assert max(a) < 1e-5
+
+    def test_bpwl_vs_inter(self):
+        a = abs(bpwl_inter(bpwl_data[0,:]).value - bpwl_data[1,:])/bpwl_data[1,:]
```

Comment:
[numpy.allclose](https://numpy.org/doc/stable/reference/generated/numpy.allclose.html) and [units.allclose](https://docs.astropy.org/en/stable/api/astropy.units.allclose.html) already perform this calculation for you. One does it with simple arrays, the other with array of units.

Reply:
Same for the following checks.